tests/raftstore/test_stale_peer.rs

```diff
@@ -62,35 +62,42 @@ fn test_stale_peer_out_of_region<T: Simulator>(cluster: &mut Cluster<T>) {
     let engine_2 = cluster.get_engine(2);
     must_get_equal(&engine_2, key, value);
 
-    // isolate peer 2 from other part of the cluster
+    // Isolate peer 2 from other part of the cluster.
     cluster.add_send_filter(IsolationFilterFactory::new(2));
 
-    // add peer [(4, 4), (5, 5), (6, 6)]
+    // Add peer [(4, 4), (5, 5), (6, 6)].
     pd_client.must_add_peer(r1, new_peer(4, 4));
     pd_client.must_add_peer(r1, new_peer(5, 5));
     pd_client.must_add_peer(r1, new_peer(6, 6));
 
-    // remove peer [(1, 1), (2, 2), (3, 3)]
+    // Remove peer [(1, 1), (2, 2), (3, 3)].
     pd_client.must_remove_peer(r1, new_peer(1, 1));
     pd_client.must_remove_peer(r1, new_peer(2, 2));
     pd_client.must_remove_peer(r1, new_peer(3, 3));
 
-    // recover peer 2 from isolation
-    cluster.clear_send_filters();
+    // Keep peer 2 isolated after peer 3 is destroyed.
+    // Otherwise if peer 3 is not destroyed, or destroyed and marked as tombstone
+    // the communication between peer 3 and peer 2 would cause peer 2 to
+    // destroy itself earlier than this test case expects
+    // due to the handling of stale message on peer 3.
```

Comment:
Keep peer 2 isolated. Otherwise, whether peer 3 is destroyed or not, it will hanle the stale message and cause peer 2 to destroy itself earlier than this test case expects.